config/initializers/content_security_policy.rb

```diff
@@ -4,17 +4,24 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-# Rails.application.config.content_security_policy do |policy|
-#   policy.default_src :self, :https
-#   policy.font_src    :self, :https, :data
-#   policy.img_src     :self, :https, :data
-#   policy.object_src  :none
-#   policy.script_src  :self, :https
-#   policy.style_src   :self, :https
+Rails.application.config.to_prepare do
+  asset_host = Rails.configuration.three_scale.asset_host.to_s.strip
 
-#   # Specify URI for violation reports
-#   # policy.report_uri "/csp-violation-report-endpoint"
-# end
+  if Rails.env.test?
```

Comment:
why use a different policy in test?

Reply:
In `test` we allow everything because `features/asset_host/asset_host.feature` fails otherwise. I explain a bit in the commit description https://github.com/3scale/porta/pull/3833/commits/83f5698602f0915312251755ffb267b56e7afa57. But basically that Cuke requires updating the CSP value on the fly between scenarios but I didn't find a way to do it.

Reply:
I don't understand the description in the PR. Can't we set that `asset_host` for tests? In current implementation two things concern me:

- we have `if test` in production code
- (more important) we don't test whether our pages properly open with the CSP. Basically we don't test the actual CSP. In 6a508b4 I tried to make sure that change makes sense for tests as well production for the same reason. I think at the end it resulted in a potentially more resilient system.